### PR TITLE
Restructure thumbor-plugins

### PR DESCRIPTION
**Pull request by guilhermef**
The current repo relies on every optimizer to be installed to run the tests.

The plan is to have a Dockerfile on each optimizer to isolate the environment and prevent interaction between it.

### CODE REVIEW

**Comment by heynemann** [comment]
Wow!!! Good job!!!

File: thumbor_plugins/optimizers/mozjpeg/setup.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    license='MIT',
+    description='Thumbor optimizer to run mozjpeg',
+    classifiers=[
+        'Development Status :: 4 - Beta',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: MIT License',
+        'Natural Language :: English',
+        'Operating System :: MacOS',
+        'Operating System :: POSIX :: Linux',
+        'Programming Language :: Python :: 2.7',
```
Comment (marcelometal) [inline thumbor_plugins/optimizers/mozjpeg/setup.py:23]: 3.x?

Comment (heynemann) [reply thumbor_plugins/optimizers/mozjpeg/setup.py:23]: 👍

**Comment by heynemann** [comment]
Since you added WIP I converted to draft so you can work on it :D

File: thumbor_plugins/optimizers/mozjpeg/setup.py (hunk 1 above)
Comment (guilhermef) [reply thumbor_plugins/optimizers/mozjpeg/setup.py:23]: Thanks, I've missed the classifiers

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+jobs:
+  list_packages:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(find thumbor_plugins -type f -name 'setup.py' | awk -F\/ '{print $2"/"$3}' | jq -Rsc '. / "\n" - [""]')"
```
Comment (heynemann) [inline .github/workflows/release.yml:19]: You can do that? Nice!!!

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Prepare build environment
+        run: |
+          pip install -U pip
+          pip install -U build
+      - name: Build package
+        run: python -m build
+      - name: Publish a Python distribution to PyPI
```
Comment (heynemann) [inline .github/workflows/release.yml:42]: Not sure I like publishing every build @guilhermef... I think releases should have some ceremony. A tag in git, a releases readme  here in github so people can follow what changed in each version. WDYT?

Comment (heynemann) [reply .github/workflows/release.yml:42]: If you think publishing every release is useful I would add '-nightly' to the release name, so users know that this is not a stable release, but a release of the main branch.

File: README.md [hunk 4]
```diff
@@ -1,28 +1,27 @@
 # thumbor-plugins
```
Comment (heynemann) [inline README.md:1]: We can definitely give some more love to this readme :D Look at thumbor and thumbor-aws for inspiration?

File: old_optimizers/__init__.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
+from thumbor.config import Config
```
Comment (heynemann) [inline old_optimizers/__init__.py:1]: Why did you move all the optimizers to old_optimizers?

File: setup.py [hunk 6]
```diff
@@ -22,31 +22,27 @@
...
-        'Topic :: Internet :: WWW/HTTP :: Dynamic Content',
-        'Topic :: Multimedia :: Graphics :: Presentation'
+        "Development Status :: 4 - Beta",
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: MIT License",
+        "Natural Language :: English",
+        "Operating System :: MacOS",
+        "Operating System :: POSIX :: Linux",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.6",
```
Comment (heynemann) [inline setup.py:32]: 3.6 is not supported in thumbor anymore. Pillow dropped support as well.

File: setup.py [hunk 7]
```diff
@@ -22,31 +22,27 @@
...
-            'tornado==4.4',
-            'tornado-pyvows>=0.6.0',
-            'coverage',
-            'mock',
-            'colorama',
-            # https://github.com/python-pillow/Pillow/issues/2609
-            'pillow<4.2.0',
-            'nose'
+            "pytest>=6.2.5",
+            'thumbor>=7.0,<8.0',
```
Comment (heynemann) [inline setup.py:45]: I think the new way of specifying versions is `thumbor==7.*,>=7.0.6 (whatever version you want RIGHT NOW). This way you can still say you want greater than 7.0.6 and don't need to worry about the range semantics (>=, < etc). More of a nitpick, but thought of sharing.

File: thumbor_plugins/__init__.py [hunk 8]
```diff
@@ -5,4 +5,4 @@
 # http://www.opensource.org/licenses/mit-license
 # Copyright (c) 2011 globo.com timehome@corp.globo.com
 
-__version__ = "0.2.4"
+__version__ = "1.0.0"
```
Comment (heynemann) [inline thumbor_plugins/__init__.py:8]: Maybe wait until we finish moving optimizers from old_optimizers to optimizers before 1.0.0?

**Review by heynemann — CHANGES_REQUESTED** [review]
Added  a bunch of comments :)

File: thumbor_plugins/__init__.py (hunk 8 above)
Comment (guilhermef) [reply thumbor_plugins/__init__.py:8]: This version doesn´t matter anymore, now all the plugins are released with thumbor-plugins-<optimizer>.

File: old_optimizers/__init__.py (hunk 5 above)
Comment (guilhermef) [reply old_optimizers/__init__.py:1]: I still need to move one by one to the new format.

File: .github/workflows/release.yml (hunk 3 above)
Comment (guilhermef) [reply .github/workflows/release.yml:42]: Technically, it will not publish on every build unless the version has changed.
The PyPI upload will ignore if the version is already published.

Comment (heynemann) [reply .github/workflows/release.yml:42]: But then how will we know what code corresponds to the version in pypi?

Comment (guilhermef) [reply .github/workflows/release.yml:42]: I'll add an action to create the tag automatically
